department-of-veterans-affairs/caseflow
# Fix bug on hearing details page where email error messages weren't being displayed

Resolves #14990 

### Description

 - Update server to throw if there are validation errors (`update => update!`)
 - Update server to handle setting the representative email to null
   - Updated server to not send down virtual hearing notification alerts if only the representative email was updated
 - Small fixes to remove PropType warnings
 - Add tests for email validations on the hearing details page

### Acceptance Criteria
- [x] Ensure validations display when modifying any email address on the hearing details page for a virtual hearing

### Testing Plan
1. Login as a Hearing Coordinator
2. Go to the hearing details page for a virtual hearing
3. Try changing representative email to `123456`
4. Try deleting the representative email
5. Try changing veteran email *and* deleting the representative email
6. Try changing the representative timezone *and then* deleting the representative email (the timezone should revert)
7. Try making both the veteran email and representative email invalid

### User Facing Changes

<img width="839" alt="Screen Shot 2020-08-25 at 10 44 44 AM" src="https://user-images.githubusercontent.com/1298274/91189069-ff06d000-e6bf-11ea-8450-e13f8c7e33bc.png">
